### ISSUE TITLE
java.lang.ClassFormatError: Invalid pc in LineNumberTable in class file

### ISSUE THREAD

**Issue by boris-petrov** [Guardsquare/proguard#15]
We include Groovy in our build and updating its version from `2.5.4` to `2.5.5` (and any above) leads to the following exception when we run the build:

```
Exception in thread "main" java.lang.ClassFormatError: Invalid pc in LineNumberTable in class file com/company/util/d
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:151)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:823)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:721)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:644)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at com.company.util.DbTools.a(SourceFile:204)
	at com.company.util.DbTools.main(SourceFile:114)
```

Searching through the "old" issue tracker I found just [this similar issue](https://sourceforge.net/p/proguard/bugs/768/), not sure if it is related.

I found this because we used the Groovy version that comes with Gradle and they updated it from `2.5.4` to `2.5.8` in Gradle 6 which broke the builds we make.

This is kind of a show-stopper for us because Groovy < `2.5.7` doesn't work with Java 13.

I could try creating a reproduction if you can't figure out what's going on.

**Comment by netomi** [comment]
Thanks for the report and referencing the ticket @ sourceforge.

I inspected it already and it looks like that some instructions (nop) are removed, but the LineNumberAttribute is not updated correctly. We will try to reproduce and fix it asap.

**Comment by netomi** [comment]
I tried to reproduce the issue with the attached class files in the other ticket, but the LineNumberAttribute gets correctly updated when using 6.2.0 with optimization and shrinking enabled.

Could you share your used configuration?

**Comment by boris-petrov** [comment]
@netomi - thanks for the support!

I was just typing an answer and I saw your edit. Yes, we use both `dontshrink` and `dontoptimize`. Without the `dontoptimize` part we were getting some errors, I'll try tomorrow and perhaps open a new issue.

Thank you again for looking into this!

**Comment by netomi** [comment]
Ok we have found the problem. Preverification removes some unused instructions that seem to be added by this new version of groovy.

So you could add -dontpreverify as a workaround, but we will fix this asap and release 6.2.1.
